rimage/image_file.go

```diff
@@ -34,6 +36,41 @@ var RGBABitmapMagicNumber = []byte("RGBA")
 // in bytes. See above as to why.
 const RawRGBAHeaderLength = 12
 
+func init() {
+	// Here we register the custom format above so that we can simply use image.Decode
+	// so long as the raw RGBA data has the appropriate header
+	image.RegisterFormat("vnd.viam.rgba", string(RGBABitmapMagicNumber),
+		func(r io.Reader) (image.Image, error) {
+			rawBytes, err := ioutil.ReadAll(r)
+			if err != nil {
+				return nil, err
+			}
+			header := rawBytes[:RawRGBAHeaderLength]
+			width := int(binary.BigEndian.Uint32(header[4:8]))
+			height := int(binary.BigEndian.Uint32(header[8:12]))
+			img := image.NewNRGBA(image.Rect(0, 0, width, height))
+			imgBytes := rawBytes[RawRGBAHeaderLength:]
+			img.Pix = imgBytes
+			return img, nil
+		},
+		func(r io.Reader) (image.Config, error) {
+			imgBytes := make([]byte, RawRGBAHeaderLength)
+			_, err := r.Read(imgBytes)
```

Comment:
We don't want to read the bytes of the entire image, just the number up to RawRGBAHeaderLength. Can you check that it only returns the minimum number of bytes necessary?

Reply:
does Read not use the capacity of the byte array to determine how many bytes to read?

Reply:
I looked it up, it [does](https://pkg.go.dev/bufio#Reader.Read). However, there may be too few bytes in the original data as @tessavitabile pointed out, so I will check for that and error accordingly